scene/resources/3d/box_shape_3d.cpp

```diff
@@ -47,6 +49,163 @@ Vector<Vector3> BoxShape3D::get_debug_mesh_lines() const {
 	return lines;
 }
 
+Ref<ArrayMesh> BoxShape3D::get_debug_arraymesh_faces(const Color &p_modulate) const {
+	int prevrow, thisrow, point;
+	float x, y, z;
+
+	Vector3 start_pos = size * -0.5;
+
+	Vector<Vector3> points;
+	Vector<Color> colors;
+	Vector<int> indices;
+	point = 0;
+
+	// front + back
+	y = start_pos.y;
+	thisrow = point;
+	prevrow = 0;
+	for (int j = 0; j <= 1; j++) {
+		x = start_pos.x;
+		for (int i = 0; i <= 1; i++) {
+			// front
+			points.push_back(Vector3(x, -y, -start_pos.z)); // double negative on the Z!
+			colors.push_back(p_modulate);
+			point++;
+
+			// back
+			points.push_back(Vector3(-x, -y, start_pos.z));
+			colors.push_back(p_modulate);
+			point++;
+
+			if (i > 0 && j > 0) {
+				int i2 = i * 2;
+
+				// front
+				indices.push_back(prevrow + i2 - 2);
+				indices.push_back(prevrow + i2);
+				indices.push_back(thisrow + i2 - 2);
+				indices.push_back(prevrow + i2);
+				indices.push_back(thisrow + i2);
+				indices.push_back(thisrow + i2 - 2);
+
+				// back
+				indices.push_back(prevrow + i2 - 1);
+				indices.push_back(prevrow + i2 + 1);
+				indices.push_back(thisrow + i2 - 1);
+				indices.push_back(prevrow + i2 + 1);
+				indices.push_back(thisrow + i2 + 1);
+				indices.push_back(thisrow + i2 - 1);
+			}
+
+			x += size.x;
+		}
+
+		y += size.y;
+		prevrow = thisrow;
+		thisrow = point;
+	}
+
+	// left + right
```

Comment:
There's more like this in the same file, I'm not sure it's productive to list all of them
```suggestion
	// Left + right.
```